cmd/theila/main.go

```diff
@@ -5,57 +5,71 @@
 package main
 
 import (
-	"fmt"
-	"io/fs"
+	"context"
 	"log"
-	"net/http"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
 
-	"github.com/go-chi/chi"
-	"github.com/go-chi/chi/middleware"
+	"github.com/spf13/cobra"
 
-	"github.com/talos-systems/theila/internal/frontend"
+	"github.com/talos-systems/theila/internal/backend"
 )
 
+// rootCmd represents the base command when called without any subcommands.
+var rootCmd = &cobra.Command{
+	Use:   "theila",
+	Short: "Talos and Sidero frontend",
+	Long:  ``,
+}
+
+var rootCmdArgs struct {
+	address string
+	port    int
+}
+
 func main() {
-	log.Println("Starting app")
+	logger := log.New(os.Stdout, "", log.Lshortfile|log.Ldate|log.Lmicroseconds|log.Ltime)
 
-	router := registerRoutes()
+	logger.Println("Starting app")
 
-	port := 8080
+	server := backend.NewServer(logger, rootCmdArgs.address, rootCmdArgs.port)
 
-	log.Printf("Serving on port %d", port)
+	ctx, cancel := context.WithCancel(context.Background())
 
-	if err := http.ListenAndServe(fmt.Sprintf("0.0.0.0:%d", port), router); err != nil {
-		log.Fatal(err)
-	}
-}
+	var wg sync.WaitGroup
 
-func registerRoutes() http.Handler {
-	log.Println("Registering routes")
+	wg.Add(1)
 
-	r := chi.NewMux()
+	go func() {
+		defer wg.Done()
 
-	r.Use(middleware.Logger)
-	r.Use(middleware.Recoverer)
+		err := server.Run(ctx)
+		if err != nil {
+			logger.Fatalf("failed to run server %s", err)
+		}
+	}()
 
-	sub, err := fs.Sub(frontend.Dist, "dist")
-	if err != nil {
-		log.Fatalf("failed to get dist/frontend directory")
-	}
+	// Listen for signals and do graceful shutdown
+	sigCh := make(chan os.Signal, 1)
+	signal.Notify(sigCh, syscall.SIGINT, syscall.SIGTERM)
```

Comment:
There is a new `signal.NotifyContext` just for that pattern

Reply:
nice! Didn't know that, thanks.